_posts/2021-05-14-bring-gitops-to-your-functions-with-argocd-part-2-the-missing-piece-tekton.md

```diff
@@ -0,0 +1,612 @@
+
+To get more detail about _Interceptors_, please refer to this [link](https://tekton.dev/docs/triggers/eventlisteners/#interceptors).
+
+```bash
+$ kubectl apply -f https://storage.googleapis.com/tekton-releases/triggers/latest/interceptors.yaml
+deployment.apps/tekton-triggers-core-interceptors created
+service/tekton-triggers-core-interceptors created
+clusterinterceptor.triggers.tekton.dev/cel created
+clusterinterceptor.triggers.tekton.dev/bitbucket created
+clusterinterceptor.triggers.tekton.dev/github created
+clusterinterceptor.triggers.tekton.dev/gitlab created
+```
+
+Verify if everything is working before moving onto the next step.
+
+```bash
+$ kubectl get pods --namespace tekton-pipelines
+NAME                                                 READY   STATUS    RESTARTS   AGE
+tekton-pipelines-controller-5cf4d4867f-s6tqv         1/1     Running   0          5m21s
+tekton-pipelines-webhook-57bfb4b4d7-7jp8c            1/1     Running   0          5m20s
+tekton-triggers-controller-7cbd49fbb8-d2bgj          1/1     Running   0          4m40s
+tekton-triggers-core-interceptors-5d7f674ccc-d4fq7   1/1     Running   0          2m5s
+tekton-triggers-webhook-748fb7778c-mpg94             1/1     Running   0          4m40s
+```
+
+### 4. Install Tasks, Pipelines and Triggers
+
+Tekton provides as some kind of building blocks as _Custom Resources_ in order to build Cloud Native CI/CD pipelines, and these are the basic ones _Tasks_ and _Pipelines_.  Task defines steps that needs to be executed. A Task is effectively a Pod, while each step is a container within that Pod. Pipeline takes the name(s) and order of execution of TaskRun object(s). A Trigger specifies what happens when the EventListener detects an event. A Trigger specifies a TriggerTemplate, a TriggerBinding, and optionally an Interceptor.
+
+To get more detail about them, please refer to this [link](https://tekton.dev/docs/pipelines/) for Tekton Primitives and this [link](https://tekton.dev/docs/triggers/) for Tekton Triggers.
+
+Let's install them but we need to first clone the repository that involves manifest files for them.
+
+```bash
+$ git clone https://github.com/developer-guy/manage-your-functions-based-on-cloud-native-ci-cd-using-tekton.git
+Cloning into 'manage-your-functions-based-on-cloud-native-ci-cd-using-tekton'...
+remote: Enumerating objects: 102, done.
+remote: Counting objects: 100% (102/102), done.
+remote: Compressing objects: 100% (75/75), done.
+remote: Total 102 (delta 47), reused 69 (delta 25), pack-reused 0
+Receiving objects: 100% (102/102), 2.49 MiB | 5.02 MiB/s, done.
+Resolving deltas: 100% (47/47), done.
+```
+
+Before installing them we do some kind of initial set up for the pipelines such as granting necessary permissions by creating RBAC, creating ssh-key to be able commit&push to the repository, creating git source as PersistentVolumeClaim, creating a secrets which contains dockerhub credentials, and ssh-key information etc.
```

Comment:
I'd rather see you split up each of these sections and explain each one before giving the command. There's a lot going on here, so give the reader a chance to absorb it.

dockerhub => Docker Hub
commit&push => commit and push